### PR TITLE
Add GSoC ideas and instructions page for 2017.

### PR DESCRIPTION
**Pull request by brainwane**
Also I've created https://groups.google.com/forum/#!forum/zulip-gsoc .

### CODE REVIEW

**Comment by brainwane** [comment]
It would be lovely to actually have the page be nicely formatted HTML and available at zulip.org. If that version works for you then I'll delete the Markdown version, and link to the HTML version in the main Zulip README.

File: gsoc-ideas.md [hunk 1]
```diff
@@ -1,40 +1,305 @@
...
+  [Haskell](https://hackage.haskell.org/package/hzulip)...
+
+### Mobile projects (mobile experience expected)
+
+Code: [github.com/zulip/zulip-mobile (experimental, cross-platform
+React Native mobile app)](https://github.com/zulip/zulip-mobile),
+[github.com/zulip/zulip-android (Android app -- a Gradle project
+written in Java)](https://github.com/zulip/zulip-android),
+[github.com/zulip/zulip-ios (iOS app -- written in
+Objective-C)](https://github.com/zulip/zulip-ios)
```
Comment (timabbott) [inline gsoc-ideas.md]: zulip-ios is a dead project, we should unlist it here.

File: gsoc-ideas.md [hunk 2]
```diff
@@ -1,40 +1,305 @@
...
+  postgresql experience desired!
+
+- Build a federation system for users on different Zulip servers to
+  exchange messages.
+
+- Write more API client libraries in more languages, or improve the
+  ones that already exist (in
+  [JavaScript](https://github.com/zulip/zulip-js),
+  [PHP](https://packagist.org/packages/mrferos/zulip-php),
+  [Haskell](https://hackage.haskell.org/package/hzulip)...
```
Comment (timabbott) [inline gsoc-ideas.md:241]: Woah, there's a haskell package?

File: gsoc-ideas.md [hunk 3]
```diff
@@ -1,40 +1,305 @@
...
+2) you ask your question
+3) someone directs you to a document
+4) you go read that document, and try to use it to answer your question
+5) you find you are confused about a new thing
+6) you ask another question
+7) now that you have demonstrated that you have the ability to read,
+think, and learn new things, someone has a longer talk with you to
+answer your new specific question
+8) you and the other person collaborate to improve the document that you
+read in step 3 :-)
```
Comment (timabbott) [inline gsoc-ideas.md:141]: This list doesn't render right.

**Review by timabbott — COMMENTED** [review]
Cool, thanks for working on this @brainwane!  I think we should keep this as Markdown, but we will probably be moving zulip.org to a hosting setup that supports rendering the markdown to HTML somewhere other than on GitHub.

File: gsoc-ideas.md (hunk 3 above)
Comment (brainwane) [reply gsoc-ideas.md:141]: Darn it. Thanks. Fixed.

File: gsoc-ideas.md (hunk 2 above)
Comment (brainwane) [reply gsoc-ideas.md:241]: I know! By Pedro Tacla Yamada, a Recurser.

**Comment by brainwane** [comment]
@timabbott Updated in response to your comments and ready for review.

(Note to my future self: `pandoc -o gsoc-ideas.html gsoc-ideas.md` works, but you'll want to search-and-replace the header tags to change the header magnitudes, and separate them with `<section></section>` divisions.)

**Comment by timabbott** [comment]
Merged, thanks @brainwane!

**Pull request closed without merging**